batch-jsr352/core/src/main/java/org/hibernate/search/batch/jsr352/core/massindexing/step/lucene/impl/EntityWriter.java

```diff
@@ -82,6 +83,7 @@ public void open(Serializable checkpoint) throws Exception {
 		searchMapping = Search.mapping( emf );
 		mappingContext = (BatchMappingContext) searchMapping;
 		typeIdentifier = mappingContext.typeContextProvider().typeIdentifierForEntityName( entityName );
+		workspace = searchMapping.scope( typeIdentifier.javaClass(), entityName ).workspace( tenantId );
```

Comment:
FYI (no need to change it): you can pass `Object.class` instead of `typeIdentifier.javaClass()` here: it's only used to check that entities implement the given class, so we don't really care here.